site/quorum-queue.md

```diff
@@ -0,0 +1,267 @@
+
+If a quorum of nodes cannot be recovered (say if 2 out of 3 RabbitMQ nodes are
+permanently lost) the queue is permanently unavailable and
+will need to be force deleted and recreated.
+
+
+## [Features compared to other queues](#feature-comparison)
+
+
+
+| Feature | Classic / HA | Quorum |
+| :-------- | :------- | ------ |
+| Non-durable queues | yes | no |
+| Exclusive queues | yes | no |
+| Persistence | per message | always |
+| Membership changes | automatic | manual  |
+| TTL | yes | no |
+| Queue size limits | yes | no |
+| Lazy | yes | no |
+| Priority | yes | no |
+| Dead letter queues | yes | yes |
+| Adheres to policies | yes | no |
+| Reacts to memory alarm | yes | no |
+
+
+## [Primary use case](#use-case)
+
+Quorum queues are _not_ designed to be used everywhere. Their intended use is for
+topologies where queues exist for a long time and are critical to some business
+facility where fault
+tolerance and data safety is more important than, say, low latency and advanced
+queue features.
+Examples would be incoming orders in a sales system or votes cast in an
+election system where potentially losing messages would have a significant
+business impact.
+
+Stock tickers and instant messaging systems benefit less or not at all from
+quorum queues.
+
+Publishers should use publisher confirms as this is how clients can interact with
+the quorum queue consensus system. Publisher confirms will only be issued once
+a published message
+has been successfully replicated to a quorum of nodes and is considered "safe"
+within the context of the system.
+
+Consumers should use manual acknowledgements to ensure messages that aren't
+successfully processed are returned to the queue so that
+another consumer can re-attempt processing.
+
+## [When not to use it](#when-not-to-use-it)
+
+When requiring:
+
+* Low latency
+* Transient or exclusive queue features.
+* Very long queue backlogs (all nodes keep all messages in memory at all times).
+* When not using manual acks and publisher confirms.
+
+
+## [Semantics under failure](#failure-semantics)
+
+The quorum queue requires a quorum of the declared
+nodes to be available to function.
+When a RabbitMQ node hosting a current quorum queue "leader" (in Raft parlance)
+crashes or is stopped another node hosting a quorum queue "follower" will
+be elected leader and resume operations. In contrast with classic HA
+queues there is no explicit synchronization of "followers" (mirrors) following
+a leader change
+which means that availability should be resumed very shortly after the failure
+event even when the queue has a significant backlog.
+
+## [Performance](#performance)
+
+Quorum queues are designed to trade latency for throughput and have been tested
+and compared against HA queues in 3, 5 and 7 node configurations at several
+message sizes. In all scenarios quorum queues have been observed to be have
+higher or equal throughput to HA queues. When message sizes are small they can
```

Comment:
This is a bold statement. Some users can see that as a promise. Is there some benchmark to reproduce with specific conditions description?

Reply:
"have been observed" does not equal any kind of promise.

Reply:
I worry about "In all scenarios" statement. Can we still produce specific scenarios description?

Reply:
"all tested scenarios"? I don't want to put specifics in a generic doc that will needs to be relevant as the feature evolves. We can also include a section on how the users will need to compare their own specific scenarios to properly evaluate performance.

Reply:
"In all tested scenarios" sounds reasonable. We can also say "most scenarios", too.